### PR TITLE
airbyte-ci: connector progressive rollout finalize / rollback steps + gh actions

### PR DESCRIPTION
**Pull request by natikgadzhi**
## What

This PR is #44877 and #44885 in a trenchcoat. They're rebased over master and I cleaned up conflicts in airbyte-ci.

They are both approved individually.

The whole thing depends on #45867 (and includes it in this PR because I'm not good at stacking), so we can just merge this.

## How

```bash
git cherry-pick 5e7cd4145120f256a9781b959f3dd8bc21a49190
git cherry-pick e02a46939abb4c85d1c823852f265ff3814868f8
```

## TODO
- [ ] I've merged @bnchrch suggestion on supporting --force in metadata task, this should be in airbyte-ci I suppose.
- [x] Feedback from @clnoll below
- [ ] Merge and _test_ on source-faker

> [!warning]
> Auto-merge on

### CODE REVIEW

File: airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/publish/commands.py
Comment (clnoll) [inline airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/publish/commands.py:148]: I won't block on this but am still wondering why we're preferring flags for `--promote-release-candidate` and `--rollback-release-candidate` instead of enforcing enums that have `PUBLISH` as the default value.

I know this was @alafanechere 's code originally but if you have an opinion @natikgadzhi I'd be interested!

**Review by clnoll — COMMENTED** [review]
Thanks @natikgadzhi , just one question. I'll approve once the comments from https://github.com/airbytehq/airbyte/pull/45867 are addressed.

File: airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/publish/commands.py
Comment (natikgadzhi) [reply airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/publish/commands.py:148]: I will take a look!

File: airbyte-ci/connectors/metadata_service/lib/metadata_service/gcs_upload.py
Comment (natikgadzhi) [inline airbyte-ci/connectors/metadata_service/lib/metadata_service/gcs_upload.py:40]:
```suggestion
from pydash import set_
from pydash.objects import get
```

File: airbyte-ci/connectors/metadata_service/lib/metadata_service/gcs_upload.py
Comment (natikgadzhi) [inline airbyte-ci/connectors/metadata_service/lib/metadata_service/gcs_upload.py:38]:
```suggestion
from pydash.objects import get

```

File: airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/publish/commands.py
Comment (natikgadzhi) [reply airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/publish/commands.py:148]: Ok, so, if I understand you correctly, we could do the following: 

1. Implement separate commands for each operation (publish, promote, rollback) — would reuse a lot of code. 
2. Instead of `--promote` and `--rollback`, we could have another option of type enum, i.e. `--operation={publish,promote, rollback}` and default this to publishing. This feels sane! About the same amount of code, but by design can never be both ;) 
3. The design that we have here in this PR already.

I originally thought it's suspicious that you can "promote" or "rollback" a few connectors at the same time, but I do not see a reason to restrict that. 

I think both designs are workable, so I'll continue reading the rest of the PR and prefer pushing forward and verifying this works, I expect we might want a few changes to this in the coming days, so I'll leave it for later.

**Comment by natikgadzhi** [comment]
I've reviewed what we had there, and I'd like to push this forward and start testing. Caveat is that I did not add a --force option, but I don't think that at _this_ stage we absolutely have to have one.

File: airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/publish/commands.py
Comment (clnoll) [reply airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/publish/commands.py]: Yes, 2 is what I had in mind. I'm good with punting on the final decision (and they do all seem workable).

File: airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/publish/context.py
Comment (bnchrch) [inline airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/publish/context.py:20]: 💎

File: airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/metadata/pipeline.py
Comment (bnchrch) [inline airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/metadata/pipeline.py:101]: 💎 Love to see `SimpleDockerStep` being used

**Review by bnchrch — APPROVED** [review]
LGTM!

Rollout Ludacris approved (but tooling should have final say :P )
a-man-with-his-mouth-open-is-surrounded-by-people-and-the-word-rollout-is-above-him

**Pull request merged**
